Strip domain component from username?

Before I dig too far into the source, I want to know if there's already a provision to strip the domain component from the username portion of the shell.

So I'm clear: right now because I'm on a Linux VM joined to an AD domain, my username shows up as username@DOMAIN.COM in my shell prompt.  That obviously takes up a lot of space and I'd like to get that back.

Thanks!

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25125845-strip-domain-component-from-username?utm_campaign=plugin&utm_content=tracker%2F283196&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F283196&utm_medium=issues&utm_source=github).
</bountysource-plugin>

I must have really hosed things up.  On my work laptop (that is actually on a domain) the env test works, but at home it's comparing my actual environment to the test environment, and that can't be right.

After running:

python -m unittest discover tests

I get this (among other failures):


======================================================================
FAIL: test_user (test_segments.TestEnv)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kodak/development/wtf/powerline/tests/test_segments.py", line 491, in test_user
    {'contents': 'def@DOMAIN.COM', 'highlight_groups': ['user']}
AssertionError: Lists differ: [{u'contents': 'kodak', u'high... != [{u'contents': u'def@DOMAIN.CO...

First differing element 0:
{u'contents': 'kodak', u'highlight_groups': [u'user']}
{u'contents': u'def@DOMAIN.COM', u'highlight_groups': [u'user']}

- [{u'contents': 'kodak', u'highlight_groups': [u'user']}]
?                 -- ^^

+ [{u'contents': u'def@DOMAIN.COM', u'highlight_groups': [u'user']}]
?                +  ^^^^^^^^^^^^^


======================================================================

--- a/powerline/segments/common/env.py
+++ b/powerline/segments/common/env.py
@@ -148,11 +148,13 @@ username = False
 _geteuid = getattr(os, 'geteuid', lambda: 1)
 
 
-def user(pl, hide_user=None):
+def user(pl, hide_user=None, hide_domain=False):
 	'''Return the current user.
 
 	:param str hide_user:
 		Omit showing segment for users with names equal to this string.
+	:param bool hide_domain:
+		Drop domain component if it exists in a username (delimited by '@').
 
 	Highlights the user with the ``superuser`` if the effective user ID is 0.
 
@@ -166,6 +168,11 @@ def user(pl, hide_user=None):
 		return None
 	if username == hide_user:
 		return None
+	if hide_domain:
+		try:
+			username = username[:username.index('@')]
+		except ValueError:
+			pass
 	euid = _geteuid()
 	return [{
 		'contents': username,

--- a/tests/test_segments.py
+++ b/tests/test_segments.py
@@ -496,15 +496,15 @@ class TestEnv(TestCommon):
 				pass
 
 			def username(self):
-				return 'def'
+				return 'def@DOMAIN.COM'
 
 			if hasattr(self.module, 'psutil') and not callable(self.module.psutil.Process.username):
 				username = property(username)
 
 		struct_passwd = namedtuple('struct_passwd', ('pw_name',))
 		new_psutil = new_module('psutil', Process=Process)
-		new_pwd = new_module('pwd', getpwuid=lambda uid: struct_passwd(pw_name='def'))
-		new_getpass = new_module('getpass', getuser=lambda: 'def')
+		new_pwd = new_module('pwd', getpwuid=lambda uid: struct_passwd(pw_name='def@DOMAIN.COM'))
+		new_getpass = new_module('getpass', getuser=lambda: 'def@DOMAIN.COM')
 		pl = Pl()
 		with replace_attr(self.module, 'pwd', new_pwd):
 			with replace_attr(self.module, 'getpass', new_getpass):
@@ -512,12 +512,18 @@ class TestEnv(TestCommon):
 					with replace_attr(self.module, 'psutil', new_psutil):
 						with replace_attr(self.module, '_geteuid', lambda: 5):
 							self.assertEqual(self.module.user(pl=pl), [
-								{'contents': 'def', 'highlight_groups': ['user']}
+								{'contents': 'def@DOMAIN.COM', 'highlight_groups': ['user']}
 							])
 							self.assertEqual(self.module.user(pl=pl, hide_user='abc'), [
+								{'contents': 'def@DOMAIN.COM', 'highlight_groups': ['user']}
+							])
+							self.assertEqual(self.module.user(pl=pl, hide_domain=False), [
+								{'contents': 'def@DOMAIN.COM', 'highlight_groups': ['user']}
+							])
+							self.assertEqual(self.module.user(pl=pl, hide_user='def@DOMAIN.COM'), None)
+							self.assertEqual(self.module.user(pl=pl, hide_domain=True), [
 								{'contents': 'def', 'highlight_groups': ['user']}
 							])
-							self.assertEqual(self.module.user(pl=pl, hide_user='def'), None)
 						with replace_attr(self.module, '_geteuid', lambda: 0):
 							self.assertEqual(self.module.user(pl=pl), [
 								{'contents': 'def', 'highlight_groups': ['superuser', 'user']}

